lang/java/reef-runtime-standalone/src/main/java/org/apache/reef/runtime/standalone/driver/NodeListManager.java

```diff
@@ -0,0 +1,304 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Management module for remote nodes in standalone runtime.
+ */
+public final class NodeListManager {
+  private static final Logger LOG = Logger.getLogger(NodeListManager.class.getName());
+
+  /**
+   * Map from containerID -> Container.
+   */
+  private final Map<String, Container> containers = new HashMap<>();
+
+  private final Set<String> nodeInfoSet;
+  private Iterator<String> nodeSetIterator;
+  private final ConfigurationSerializer configurationSerializer;
+  private final REEFFileNames fileNames;
+  private final double jvmHeapFactor;
+  private final REEFEventHandlers reefEventHandlers;
+  private final String errorHandlerRID;
+  private final String rootFolder;
+  private final String nodeFolder;
+  private final int sshPort = 22;
+
+  @Inject
+  NodeListManager(@Parameter(NodeInfoSet.class) final Set<String> nodeInfoSet,
+                  final ConfigurationSerializer configurationSerializer,
+                  final REEFFileNames fileNames,
+                  @Parameter(JVMHeapSlack.class) final double jvmHeapSlack,
+                  final RemoteManager remoteManager,
+                  final REEFEventHandlers reefEventHandlers,
+                  @Parameter(RootFolder.class) final String rootFolder,
+                  @Parameter(NodeFolder.class) final String nodeFolder) {
+    this.nodeInfoSet = nodeInfoSet;
+    this.configurationSerializer = configurationSerializer;
+    this.fileNames = fileNames;
+    this.jvmHeapFactor = 1.0 - jvmHeapSlack;
+    this.reefEventHandlers = reefEventHandlers;
+    this.errorHandlerRID = remoteManager.getMyIdentifier();
+    this.rootFolder = rootFolder;
+    this.nodeFolder = nodeFolder;
+
+    this.nodeSetIterator = this.nodeInfoSet.iterator();
+
+    remoteManager.registerHandler(ReefServiceProtos.RuntimeErrorProto.class,
+        new EventHandler<RemoteMessage<ReefServiceProtos.RuntimeErrorProto>>() {
+          @Override
+          public void onNext(final RemoteMessage<ReefServiceProtos.RuntimeErrorProto> value) {
+            final FailedRuntime error = new FailedRuntime(value.getMessage());
+            LOG.log(Level.SEVERE, "FailedRuntime: " + error, error.getReason().orElse(null));
+            release(error.getId());
+          }
+        });
+
+    LOG.log(Level.FINEST, "Initialized NodeListManager.");
+  }
+
+  void release(final String containerID) {
+    synchronized (this.containers) {
+      final Container ctr = this.containers.get(containerID);
+      if (null != ctr) {
+        LOG.log(Level.INFO, "Releasing Container with containerId [{0}]", ctr);
+        if (ctr.isRunning()) {
+          ctr.close();
+        }
+        this.containers.remove(ctr.getContainerID());
+      } else {
+        LOG.log(Level.INFO, "Ignoring release request for unknown containerID [{0}]", containerID);
+      }
+    }
+  }
+
+  public void onResourceLaunchRequest(final ResourceLaunchEvent resourceLaunchEvent) {
```

Comment:
@johnyangk It's actually derived from the `local/driver/ResourceManager` class. Should we keep consistency or change it?